diff --git a/src/pip/_vendor/pkg_resources/__init__.py b/src/pip/_vendor/pkg_resources/__init__.py
index a457ff27..31331a31 100644
--- a/src/pip/_vendor/pkg_resources/__init__.py
+++ b/src/pip/_vendor/pkg_resources/__init__.py
@@ -562,6 +562,8 @@ class WorkingSet:

         if entries is None:
             entries = sys.path
+            if '' in entries:
+                entries.remove('')

         for entry in entries:
             self.add_entry(entry)
