diff --git a/src/pip/_vendor/certifi/core.py b/src/pip/_vendor/certifi/core.py
index 497d938d0..60ad982c6 100644
--- a/src/pip/_vendor/certifi/core.py
+++ b/src/pip/_vendor/certifi/core.py
@@ -33,7 +33,7 @@ def where() -> str:
             # We also have to hold onto the actual context manager, because
             # it will do the cleanup whenever it gets garbage collected, so
             # we will also store that at the global level as well.
-            _CACERT_CTX = get_path("certifi", "cacert.pem")
+            _CACERT_CTX = get_path("pip._vendor.certifi", "cacert.pem")
             _CACERT_PATH = str(_CACERT_CTX.__enter__())

         return _CACERT_PATH
@@ -65,4 +65,4 @@ def where() -> str:


 def contents() -> str:
-    return read_text("certifi", "cacert.pem", encoding="ascii")
+    return read_text("pip._vendor.certifi", "cacert.pem", encoding="ascii")
