diff --git a/src/pip/_vendor/platformdirs/__init__.py b/src/pip/_vendor/platformdirs/__init__.py
index 693b64843..07baa5b1d 100644
--- a/src/pip/_vendor/platformdirs/__init__.py
+++ b/src/pip/_vendor/platformdirs/__init__.py
@@ -17,13 +17,13 @@ from .version import __version__, __version_info__
 
 def _set_platform_dir_class() -> Type[PlatformDirsABC]:
     if os.getenv("ANDROID_DATA") == "/data" and os.getenv("ANDROID_ROOT") == "/system":
-        module, name = "platformdirs.android", "Android"
+        module, name = "pip._vendor.platformdirs.android", "Android"
     elif sys.platform == "win32":
-        module, name = "platformdirs.windows", "Windows"
+        module, name = "pip._vendor.platformdirs.windows", "Windows"
     elif sys.platform == "darwin":
-        module, name = "platformdirs.macos", "MacOS"
+        module, name = "pip._vendor.platformdirs.macos", "MacOS"
     else:
-        module, name = "platformdirs.unix", "Unix"
+        module, name = "pip._vendor.platformdirs.unix", "Unix"
     result: Type[PlatformDirsABC] = getattr(importlib.import_module(module), name)
     return result
 
