diff --git a/src/pip/_vendor/urllib3/contrib/emscripten/__init__.py b/src/pip/_vendor/urllib3/contrib/emscripten/__init__.py
index 8a3c5bebd..b3c206e90 100644
--- a/src/pip/_vendor/urllib3/contrib/emscripten/__init__.py
+++ b/src/pip/_vendor/urllib3/contrib/emscripten/__init__.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-import urllib3.connection
+import urllib3.connection as urllib3_connection
 
 from ...connectionpool import HTTPConnectionPool, HTTPSConnectionPool
 from .connection import EmscriptenHTTPConnection, EmscriptenHTTPSConnection
@@ -12,5 +12,5 @@ def inject_into_urllib3() -> None:
     # if it isn't ignored
     HTTPConnectionPool.ConnectionCls = EmscriptenHTTPConnection
     HTTPSConnectionPool.ConnectionCls = EmscriptenHTTPSConnection
-    urllib3.connection.HTTPConnection = EmscriptenHTTPConnection  # type: ignore[misc,assignment]
-    urllib3.connection.HTTPSConnection = EmscriptenHTTPSConnection  # type: ignore[misc,assignment]
+    urllib3_connection.HTTPConnection = EmscriptenHTTPConnection  # type: ignore[misc,assignment]
+    urllib3_connection.HTTPSConnection = EmscriptenHTTPSConnection  # type: ignore[misc,assignment]
diff --git a/src/pip/_vendor/urllib3/contrib/pyopenssl.py b/src/pip/_vendor/urllib3/contrib/pyopenssl.py
index b89a6dab8..407d2cb30 100644
--- a/src/pip/_vendor/urllib3/contrib/pyopenssl.py
+++ b/src/pip/_vendor/urllib3/contrib/pyopenssl.py
@@ -28,8 +28,8 @@ like this:
 .. code-block:: python
 
     try:
-        import urllib3.contrib.pyopenssl
-        urllib3.contrib.pyopenssl.inject_into_urllib3()
+        import urllib3.contrib.pyopenssl as pyopenssl
+        pyopenssl.inject_into_urllib3()
     except ImportError:
         pass
 
diff --git a/src/pip/_vendor/urllib3/http2.py b/src/pip/_vendor/urllib3/http2.py
index ceb40602d..60315bd89 100644
--- a/src/pip/_vendor/urllib3/http2.py
+++ b/src/pip/_vendor/urllib3/http2.py
@@ -8,8 +8,8 @@ import h2.config  # type: ignore[import-untyped]
 import h2.connection  # type: ignore[import-untyped]
 import h2.events  # type: ignore[import-untyped]
 
-import urllib3.connection
-import urllib3.util.ssl_
+import urllib3.connection as urllib3_connection
+import urllib3.util.ssl_ as urllib3_util_ssl_
 from urllib3.response import BaseHTTPResponse
 
 from ._collections import HTTPHeaderDict
@@ -217,14 +217,14 @@ class HTTP2Response(BaseHTTPResponse):
 
 def inject_into_urllib3() -> None:
     HTTPSConnectionPool.ConnectionCls = HTTP2Connection
-    urllib3.connection.HTTPSConnection = HTTP2Connection  # type: ignore[misc]
+    urllib3_connection.HTTPSConnection = HTTP2Connection  # type: ignore[misc]
 
     # TODO: Offer 'http/1.1' as well, but for testing purposes this is handy.
-    urllib3.util.ssl_.ALPN_PROTOCOLS = ["h2"]
+    urllib3_util_ssl_.ALPN_PROTOCOLS = ["h2"]
 
 
 def extract_from_urllib3() -> None:
     HTTPSConnectionPool.ConnectionCls = orig_HTTPSConnection
-    urllib3.connection.HTTPSConnection = orig_HTTPSConnection  # type: ignore[misc]
+    urllib3_connection.HTTPSConnection = orig_HTTPSConnection  # type: ignore[misc]
 
-    urllib3.util.ssl_.ALPN_PROTOCOLS = ["http/1.1"]
+    urllib3_util_ssl_.ALPN_PROTOCOLS = ["http/1.1"]
